### PR TITLE
Add a way of setting the initial label for the default queue

### PR DESCRIPTION
**Pull request by litherum**
***
<a href="https://pr-preview.s3.amazonaws.com/litherum/gpuweb/pull/2645.html" title="Last updated on Mar 15, 2022, 8:14 PM UTC (b938754)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2645/3382f32...litherum:b938754.html" title="Last updated on Mar 15, 2022, 8:14 PM UTC (b938754)">Diff</a>

### CODE REVIEW

**Comment by kainino0x** [comment]
This strikes me as a little awkward, but otherwise fine. I suppose we could instead specify something like, the queue's initial label is always `"queue for " + device.name`?

**Comment by toji** [comment]
I agree with Kai. The functionality isn't bad, but it's an usual place to put it. Obviously there's not an alternative, however, since the queue is created with the device.

A couple of thoughts:
  - It's not clear to me what situations having a label at initialization would be necessary for? It's certainly nice to be able to set them at creation time, but I don't see it as problematic to set this label after device creation.
  - Is there a chance that we'll need to have other queue creation parameters in the future? Especially if we begin allowing creation of multiple queues. If so, then maybe we should generalize this a bit further preemptively and accept a full `GPUQueueDescriptor` when requesting the device. Something like:

```js
adapter.requestDevice({
  requiredFeatures: [],
  defaultQueue: {
    label: 'myQueue',
    theoreticalFoo: 'bar'
  },
});
```

**Comment by kainino0x** [comment]
Good thoughts, I agree with both. I'm OK with either leaving this out entirely or going ahead and adding a queue descriptor inside the device descriptor.

**Comment by litherum** [comment]
I've updated the PR to add a `GPUQueueDescriptor` which inherits from `GPUObjectBase` and has no members.

File: spec/index.bs [hunk 1]
```diff
@@ -1806,6 +1806,7 @@ interface GPUAdapter {
 dictionary GPUDeviceDescriptor : GPUObjectDescriptorBase {
     sequence<GPUFeatureName> requiredFeatures = [];
     record<DOMString, GPUSize64> requiredLimits = {};
+    GPUQueueDescriptor defaultQueueDescriptor = {};
```
[the remark this answers is not in the document]
Comment (kainino0x) [reply spec/index.bs:1809]: No opinion :) Rest LGTM

**Review by kainino0x — APPROVED** [review]

**Pull request merged**